### PR TITLE
Add flask-injector recipe

### PR DESCRIPTION
**Pull request by fuglede**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment (this one) confirming they are willing to be listed there

In addition to the above, we'll need
- [x] `injector` itself is available in conda-forge, i.e. https://github.com/conda-forge/staged-recipes/pull/12895 is merged. (As such, I hope a WIP-style PR like this one is acceptable. Otherwise please just let me know.)

One thing I was unsure about was the `typing  # [py<35]`: Injector only supports py>=35 (and depends on typing_extensions for 35<py<39), but the author of the two packages is the same, so I left it in.

### CODE REVIEW

**Comment by synapticarbors** [comment]
You'll need to get https://pypi.org/project/injector/ up on conda-forge first before this will build.

**Comment by fuglede** [comment]
> You'll need to get https://pypi.org/project/injector/ up on conda-forge first before this will build.

Ah, right, I had added a checklist point above about getting that PR merged, but that's of course not enough, until the package itself is actually available.

**Comment by fuglede** [comment]
> If you'd like to keep it open, please comment/push and we will be happy to oblige!

Yes please. Really, this one is just waiting for #12895, in which @synapticarbors had a question for @conda-forge/help-python.

**Comment by fuglede** [comment]
#12895 has been completed, so this one should be good to go as well.

File: recipes/flask-injector/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - flask
+    - injector >=0.10.0
+    - python
+    - typing  # [py<35]
```
Comment (dopplershift) [inline recipes/flask-injector/meta.yaml:25]: We're not even building packages for Python 3.5 any more. So instead can you add `noarch: python` above under `build:`, and then change the Python dependency here and under `host:` to `python >=3.6`?

Comment (dopplershift) [reply recipes/flask-injector/meta.yaml:25]: and go ahead and remove the dependency on `typing`.

**Review by dopplershift — APPROVED** [review]

**Pull request merged**